torch/distributed/pipelining/PipelineStage.py

```diff
@@ -549,6 +575,42 @@ def backward_one_chunk(
         )
         self.bwd_chunk_id += 1
 
+    def _validate_fwd_input(self, args, kwargs):
```

Comment:
the way output validation works is much cleaner since i was easily able to define the output schema as a tuple, and define a clear 'configure' and 'get' api for accessing that schema.

for inputs, the ambiguity between args and kwargs makes that a bit harder.

1. if we had a nice api for input shape like the ones for output shapes, the each Stage.__init__ would be expected to call the API like `self.configure_inputs_meta(args_meta, kwargs_meta)`.  But the user may not even provide kwargs values if they rely on defaults.  Is it gauranteed we'd get an example value for every kwarg at __init__ time?


2. users could pass different combinations of *args/**kwargs that are equally valid.
e.g. for a forward function that accepts (foo, bar=None), the user may call any of
  a) step(args=[fooval, barval], kwargs={})
  b) step(args=[fooval], kwargs={"bar": barval})
  c) step(args=[fooval])
In these cases, we can reasonbly assert that the first N values in *args will map to the first N values configured for args in configure_inputs_meta.  But its hard to reason about any additional args like in case (a)